source/base/mpi.cc

```diff
@@ -778,26 +778,9 @@ namespace Utilities
 
       // Initialize Kokkos
       {
-        // argv has argc+1 elements and the last one is a nullptr. For appending
-        // one element we thus create a new argv by copying the first argc
-        // elements, append the new option, and then a nullptr.
-        std::vector<char *> argv_new(argc + 2);
-        for (int i = 0; i < argc; ++i)
-          argv_new[i] = argv[i];
-        std::stringstream threads_flag;
-#if KOKKOS_VERSION >= 30700
-        threads_flag << "--kokkos-num-threads=" << MultithreadInfo::n_threads();
-#else
-        threads_flag << "--kokkos-threads=" << MultithreadInfo::n_threads();
-#endif
-        const std::string threads_flag_string = threads_flag.str();
-        argv_new[argc]     = const_cast<char *>(threads_flag_string.c_str());
-        argv_new[argc + 1] = nullptr;
-        // The first argument in Kokkos::initialize is of type int&. Hence, we
-        // need to define a new variable to pass to it (instead of using argc+1
-        // inline).
-        int argc_new = argc + 1;
-        Kokkos::initialize(argc_new, argv_new.data());
+        Kokkos::InitArguments args;
+        args.num_threads = MultithreadInfo::n_threads();
+        Kokkos::initialize(args);
```

Comment:
If we want to go this route we should do
```suggestion
#if KOKKOS_VERSION >= 30700
        const auto args =
          Kokkos::InitializationSettings().set_num_threads(
            MultithreadInfo::n_threads());
#else
        const Kokkos::InitArguments args(MultithreadInfo::n_threads());
#endif
        Kokkos::initialize(args);
```
Hopefully, we can update the minimal requirement for `Trilinos` soon so that our minimal `Kokkos` version is consistent.

Reply:
Thanks for the code snippet!